x-pack/plugins/cases/public/components/all_cases/utils/sanitize_filter_options.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FilterOptions } from '../../../../common/ui/types';
+import type { CaseStatuses, CaseSeverity } from '../../../../common/types/domain';
+
+const notAll = (option: string) => option !== 'all';
+
+/**
+ * In earlier versions, the options 'status' and 'severity' could have a value of 'all'.
+ * This function ensures such legacy values are removed from the URL parameters to maintain
+ * backwards compatibility.
+ */
+export const removeLegacyValuesFromOptions = ({
+  status: legacyStatus,
+  severity: legacySeverity,
+}: {
+  status: Array<CaseStatuses | 'all'>;
+  severity: Array<CaseSeverity | 'all'>;
+}): { status: CaseStatuses[]; severity: CaseSeverity[] } => {
+  return {
+    status: legacyStatus?.filter(notAll).filter(Boolean) as CaseStatuses[],
+    severity: legacySeverity?.filter(notAll).filter(Boolean) as CaseSeverity[],
+  };
+};
+
+export const getStoreableFilters = (
+  filterOptions: Partial<FilterOptions>
+): { status: CaseStatuses[] | undefined; severity: CaseSeverity[] | undefined } => {
+  const { status, severity } = filterOptions;
+
+  return { severity, status };
```

Comment:
When no `status`/`severity` filter is selected the URL will have `severity=&status=`.

<details><summary>See here.</summary><img width="1204" alt="Screenshot 2023-11-07 at 10 41 10" src="https://github.com/elastic/kibana/assets/1533137/1d3ddd7e-d8e1-42c0-8487-d2312f7c6562"></details>

I think it is because we return severity and status here even when they are an empty array.

This function is used [in here](https://github.com/elastic/kibana/pull/169371/files#diff-0e8ad282ab7faf5c8759ce565c67eb96461f3f5c43f8d7753f0a1f57389125aaR209) to generate the URL with the filtering options.

I think it would make more sense to exclude options without value from the URL.

Reply:
I tried this one but I gave up after realising that its not worth it compared agains what we gain with it. The code needs extra complexity with extra checks and filtering in several places and on the other hand, it's not an invalid url and it represents the filter status as it is